From 07a25184c5e5dd42029c6c853ab26d8cba92f474 Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Fri, 10 Oct 2014 19:47:58 +0800
Subject: [PATCH] dwc3: use USB_EVENT_NONE for cable disconnect event.

Use USB_EVENT_NONE for cable disconnect event, avoid extra paramter
for vbus status.

Change-Id: If34470023971a6d531cb9ba09b667e3948bd6b3f
Signed-off-by: Wu, Hao <hao.wu@intel.com>
(cherry picked from commit 7a2ac158311b2d42c9f081b278c19a18688d9b7b)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/dwc3/core.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index ef2829f..e3b518b 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -546,21 +546,21 @@ static int dwc3_handle_otg_notification(struct notifier_block *nb,
 	unsigned long flags;
 	int state = NOTIFY_DONE;
 	static int last_value = -1;
-	int val;
 
-	val = *(int *)data;
-
-	if (last_value == val)
+	if (last_value == event)
 		goto out;
 
-	last_value = val;
-
 	spin_lock_irqsave(&dwc->lock, flags);
 	switch (event) {
 	case USB_EVENT_VBUS:
-		dev_info(dwc->dev, "DWC3 OTG Notify USB_EVENT_VBUS, val = %d\n", val);
-		if (val)
-			pm_runtime_get(dwc->dev);
+		dev_info(dwc->dev, "DWC3 OTG Notify USB_EVENT_VBUS\n");
+		last_value = event;
+		pm_runtime_get(dwc->dev);
+		state = NOTIFY_OK;
+		break;
+	case USB_EVENT_NONE:
+		dev_info(dwc->dev, "DWC3 OTG Notify USB_EVENT_NONE\n");
+		last_value = event;
 		state = NOTIFY_OK;
 		break;
 	default:
-- 
1.7.9.5

